### PR TITLE
Refactoing component

### PR DESCRIPTION
**Pull request by fayez-baig**
refactored input fields and buttons

### CODE REVIEW

File: src/components/NxButton.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+import Button from "@material-ui/core/Button";
+
+const spacing = {
+  margin: "24px 0 16px",
+};
+const NxButton = (props) => {
+  const { type, variant, color, disabled, children } = props;
```
Comment (faran4engg) [inline src/components/NxButton.jsx:9]: @fayez-baig - children is not appropriate here. Please rename this to `btnText`

File: src/pages/Login/Login.jsx [hunk 2]
```diff
@@ -77,6 +76,7 @@ const Login = (props) => {
                   <Grid item xs={12}>
                     <NxInput
                       input={input}
+                      type="email"
```
Comment (faran4engg) [inline src/pages/Login/Login.jsx:79]: nice one 👍

File: src/pages/Login/Login.jsx [hunk 3]
```diff
@@ -87,30 +87,26 @@ const Login = (props) => {
...
               </Field>
             </Grid>
             <div className="buttons">
-              <Button
+              <NxButton
                 type="submit"
-                fullWidth
                 variant="contained"
                 color="primary"
-                className={classes.submit}
```
Comment (faran4engg) [inline src/pages/Login/Login.jsx:109]: Why did we remove `className` ? How are we managing this prop ?

File: src/pages/Signup/Signup.jsx [hunk 4]
```diff
@@ -175,41 +176,37 @@ const Signup = ({ dispatch }) => {
...
                   </Grid>
                 )}
               </Field>
             </Grid>
 
             <div className="buttons">
-              <Button
+              <NxButton
                 type="submit"
                 fullWidth
```
Comment (faran4engg) [inline src/pages/Signup/Signup.jsx:211]: Do we really need `fullWidth` prop here ?

File: src/pages/Signup/Signup.jsx [hunk 5]
```diff
@@ -72,27 +74,26 @@ const Signup = ({ dispatch }) => {
               <Field name="firstName">
                 {({ input, meta }) => (
                   <Grid item xs={12} sm={6}>
-                    <TextField
-                      {...input}
-                      variant="outlined"
+                    <NxInput
+                      input={input}
+                      id="firstname"
                       fullWidth
```
Comment (faran4engg) [inline src/pages/Signup/Signup.jsx:80]: same here for fullWidth

File: src/pages/Signup/Signup.jsx [hunk 6]
```diff
@@ -72,27 +74,26 @@ const Signup = ({ dispatch }) => {
...
               </Field>
               <Field name="lastName">
                 {({ input, meta }) => (
                   <Grid item xs={12} sm={6}>
-                    <TextField
-                      {...input}
+                    <NxInput
+                      input={input}
                       variant="outlined"
                       fullWidth
```
Comment (faran4engg) [inline src/pages/Signup/Signup.jsx:93]: here as well

File: src/pages/Signup/Signup.jsx [hunk 7]
```diff
@@ -101,29 +102,29 @@ const Signup = ({ dispatch }) => {
               <Field name="email">
                 {({ input, meta }) => (
                   <Grid item xs={12}>
-                    <TextField
-                      {...input}
+                    <NxInput
+                      input={input}
                       variant="outlined"
                       fullWidth
```
Comment (faran4engg) [inline src/pages/Signup/Signup.jsx:108]: here too

File: src/pages/Signup/Signup.jsx [hunk 8]
```diff
@@ -146,27 +147,27 @@ const Signup = ({ dispatch }) => {
               <Field name="state">
                 {({ input, meta }) => (
                   <Grid item xs={12} sm={6}>
-                    <TextField
-                      {...input}
+                    <NxInput
+                      input={input}
                       variant="outlined"
                       fullWidth
```
Comment (faran4engg) [inline src/pages/Signup/Signup.jsx:153]: Please check everywhere

File: src/components/NxButton.jsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+import Button from "@material-ui/core/Button";
+
+const spacing = {
+  margin: "24px 0 16px",
+};
```
Comment (faran4engg) [inline src/components/NxButton.jsx:7]: why this ????

File: src/pages/Login/Login.jsx (hunk 2 above)
Comment (fayez-baig) [reply src/pages/Login/Login.jsx:79]: thanks

File: src/pages/Login/Login.jsx (hunk 3 above)
Comment (fayez-baig) [reply src/pages/Login/Login.jsx:109]: updated..

File: src/components/NxButton.jsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import Button from "@material-ui/core/Button";
+
+const NxButton = (props) => {
+  const { type, variant, color, disabled, children, classProp } = props;
+  return (
+    <Button
+      className={classProp}
+      type={type}
+      fullWidth
+      variant={variant}
```
Comment (faran4engg) [inline src/components/NxButton.jsx:12]: add ternary here !

Comment (fayez-baig) [reply src/components/NxButton.jsx:12]: done

File: src/components/NxButton.jsx (hunk 1 above)
Comment (fayez-baig) [reply src/components/NxButton.jsx]: updated

File: src/pages/Signup/Signup.jsx (hunk 4 above)
Comment (fayez-baig) [reply src/pages/Signup/Signup.jsx]: nope

File: src/pages/Signup/Signup.jsx (hunk 6 above)
Comment (fayez-baig) [reply src/pages/Signup/Signup.jsx]: updated

File: src/pages/Signup/Signup.jsx (hunk 7 above)
Comment (fayez-baig) [reply src/pages/Signup/Signup.jsx]: done

File: src/pages/Signup/Signup.jsx (hunk 8 above)
Comment (fayez-baig) [reply src/pages/Signup/Signup.jsx]: done

File: src/components/NxButton.jsx (hunk 9 above)
Comment (fayez-baig) [reply src/components/NxButton.jsx]: testing purpose..

File: src/components/NxButton.jsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+const NxButton = (props) => {
+  const { type, variant, color, disabled, btnText, classProp } = props;
+  return (
+    <Button
+      className={classProp}
+      type={type}
+      fullWidth
+      variant={variant ? variant : "contained"}
+      color={color ? color : "primary"}
+      disabled={disabled}
```
Comment (faran4engg) [inline src/components/NxButton.jsx:14]: does this disable also needs to be taken care of ?

Comment (fayez-baig) [reply src/components/NxButton.jsx]: done

**Pull request merged**